### ISSUE TITLE
Improve docs on finetune mode of BatchNormalization

### ISSUE THREAD

**Issue by kmaehashi** [chainer/chainer#5292]
It is better to clearly say that `finetune` mode is the feature to reproduce the original BatchNormalization paper, and is rarely used for other purposes.
Related to #5277

**Comment by fiarabbit** [comment]
Do you mean "Even if the user has trained the BN layer with Dataset A and intend to finetune the model with Dataset B, he/she does not necessarily have to use _finetune_-mode. This is because (i) many papers used _train_-mode for finetuning, and even in the _train_-mode, the model computes the moving average for the population statistics and there is no necessity that the original population statistic is robust for finetuning."

If so, I think if the dataset size is much smaller than the original dataset and the batch statistic is suspicious for the testing dataset, the _finetune_-mode is preferable compared to the _train_-mode. (So I'm worried that the explanation might be confusing.)

**Comment by fiarabbit** [comment]
I could not find the special modification for fine-tuning in _Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift_, which is referred in `F.batch_normalization`.
Could you clarify what is _the original BatchNormalization paper_ ?

**Comment by fiarabbit** [comment]
@kmaehashi Does the _fine-tune mode_ refer _"Adaptive Batch Normalization for practical domain adaptation_"?

**Comment by fiarabbit** [comment]
OMG, this issue is re-opened. This comment

> In section 3.1. of Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift, it is mentioned.
> 
> "The normalization of activations that depends on the mini-batch allows efficient training, but is neither necessary nor desirable during inference; we want the output to depend only on the input, deterministically. For this, once the network has been trained, we use the normalization
> xhat = x−E[x] / sqrt(Var[x] + epsilon)
> using the population, rather than mini-batch, statistics"
> 
> If I understand correctly, the way to achieve that in the current implementation would be to use the fine-tuning mode.

refers the behavior in the condition where `chainer.config.train==False`. However, if I correctly understand, this issue is about the behavior when `chainer.config.train==True` and `finetune=True`. This condition is not written in the original paper, I think.
